src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -250,10 +250,16 @@ public void parse(ParseContext context) throws IOException {
         if (token == XContentParser.Token.START_ARRAY) {
             token = context.parser().nextToken();
             while (token != XContentParser.Token.END_ARRAY) {
+                if (Float.isNaN(context.parser().floatValue())) {
```

Comment:
Should we also add check for Float.isInfinite?

https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#isInfinite(float)

Reply:
Yes I think that makes sense.